jekyll/_cci2/orbs-best-practices.md

```diff
@@ -258,7 +258,62 @@ Each usage example must present a full example including showing the orb being i
 {: #secrets-should-never-be-directly-entered }
 {:.no_toc}
 
-Any information that could be considered "secret" such as API keys, auth tokens and passwords, should never be entered directly as parameter values. Instead, the orb developer should use the [env_var_name]({{site.baseurl}}/reusing-config/#environment-variable-name) parameter type, which expects the string value of the name of the environment variable that contains the secret information.
+Any information that could be considered "secret" such as API keys, auth tokens and passwords, should never be entered directly as parameter values. Instead, the orb developer should use the [env_var_name]({{site.baseurl}}/reusing-config/#environment-variable-name) parameter type, which expects the string value of the name of the environment variable that contains the secret information. The `env_var_name` parameter type will fail validation if the parameter value is not a POSIX-compliant environment variable name. This is designed to assist in preventing users from accidentally entering the secret value directly as a string.
+
+If your orb requires an "API key", you should create a parameter named `api_key` of type `env_var_name`. The user of your orb would then pass the name of the environment variable containing the API key as the value of the `api_key` parameter.
+
+```yaml
+parameters:
+  api_key:
+    type: env_var_name
+    default: API_KEY # An environment variable named API_KEY contains the API key
+steps:
+  - run:
+      environment:
+        PARAM_API_KEY: << parameters.api_key >>
+      command: |
+        # Use the API key
+```
+In this example, the string value of `API_KEY` has been saved in the environment variable `PARAM_API_KEY`. In order to get the value of the environment variable with the name `API_KEY`, we must evaluate with [BASH parameter expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html).
+
+```yaml
+steps:
+  - run:
+      environment:
+        PARAM_API_KEY: << parameters.api_key >>
+      command: |
+        API_KEY_VALUE=${!PARAM_API_KEY}
+```
+
+#### Accepting parameters as strings or environment variables
+{: #accepting-parameters-as-strings-or-environment-variables }
+{:.no_toc}
+
+When creating a parameter that accepts a non-secret string value, it is often useful to also accept an environment variable that could be reused in multiple places. For example, setting your `AWS_REGION` is likely not a secret in the traditional sense, you may want to manually entire a region, or override a default value directly. It may also be easier to reference an environment variable that contains the region, such as if you were using a [context](/docs/contexts) to store your AWS metadata for multiple projects.
```

Comment:
```suggestion
When creating a parameter that accepts a non-secret string value, it is often useful to also accept an environment variable that could be reused in multiple places. For example, setting your `AWS_REGION` is likely not a secret in the traditional sense, you may want to manually enter a region, or override a default value directly. It may also be easier to reference an environment variable that contains the region, for example, if you were using a [context](/docs/contexts) to store your AWS metadata for multiple projects.
```